docs/developer/designs/time-aware-fairness/time-aware-fairness.md

```diff
@@ -0,0 +1,196 @@
+- Applying $\max{\{,0\}}$ to the result floors negative values to 0
+
+This ensures that:
+
+$\sum{P'} = 1$
+
+$0<=P'_i<=1$
+
+For each resource division round, which makes it simpler to divide resources.
+
+### Explanation of the Formula
+
+The accounting service will provide the scheduler with data for each cycle about the usage of resources per queue per resource. A vector will represent the usage of each queue per resource type.  
+The resource division formula for over-quota resource can be found in proportion plugin, resource_division.go:
+```Go
+fairShare := amountToGiveInCurrentRound * (overQuotaWeight / totalWeights)
+```
+This calculation is done for each resource type. The result of this calculation is later capped with the requested resource of the queue, so each round of this calculation for all queues could result in remaining capacity to divide - this is run in rounds, each time with the updated remaining unsatisfied queues, until all resources are divided or all queues get their request.
+
+For convenience, we'll write it out as a formula:
+
+$$\text{F} = C \cdot \frac{W}{\sum{W}}$$
+
+Where:
+- **$F$** is the fair share allocated to a specific queue in the current round
+- **$C$** is the remaining capacity (max amount to give in current round)
+- **$w$** is the weight for a specific queue
+- **$∑w$** is the sum of weights across all competing queues
+
+Let's also mark the normalized weight as:
+
+$$W'_i = \frac{W_i}{\sum{W}}$$
+
+### Cluster-capacity normalized usage
+
+First, we'll define a normalized "usage" score for each queue and resource:
+
+$$U'_i = \frac{U_i}{T}$$
+
+Where $T$ is the total potential resource capacity for the cluster in the relevant time period. For a cluster that is fully occupied, $\sum{U_i} = T$.
+
+This works better than a naive normalization ($U'_i = \frac{U_i}{\sum{U}}$), because it doesn't result in heavy penalty when total usage ($\sum{U_i}$) is relatively low: in other words, queues are not penalized heavily for utilizing uncontended resources.
+
+
+### Penalizing queues with historical usage
+
+Now, we can adjust our fair-share calculation for resources to consider historical usage. Let's define:
+
+$$F_i = C \cdot P'_i$$
+
+Where:
+- **$F_i$** is the fair share allocated to a specific queue in the current round
+- **$C$** is the remaining capacity (max amount to give in current round)
+- **$P'_i$** is the normalized portion for queue i, defined as:
+
+$$P_i = \max{\{W'_i - k \cdot (W'_i - U'_i), 0\}}$$
+
+$$P'_i = \frac{P_i}{\sum{P}}$$
+
+Where:
+- $W'_i$ is the over-quota weight of the queue, normalized to the sum of non-satisfied queues' over-quota weight for the round
+- $\sum{P}$ considers only queues that participate in the current resource division round
+- $U'_i$ is the historical usage for the queue, normalized to the capacity of the cluster
+- $k$ is set by the admin and controls the "significance" of historical usage
```

Comment:
Is k always $0 <= k <= 1$ ? If so it means the max might not be really needed.

Reply:
Also you can't assume that u < k